src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentInvoker.cs

```diff
@@ -34,6 +38,12 @@ public void Invoke([NotNull] ViewComponentContext context)
                 context.Arguments);
             if (method == null)
             {
+                _logger.LogError(
+                    "Could not find a method with name '{ViewComponentMethodName}' on view " +
+                    "component '{ViewComponentName}'.",
+                    ViewComponentMethodSelector.SyncMethodName,
+                    context.ViewComponentDescriptor.FullName);
```

Comment:
Discussed offline with @rynowak and @lodejard : This is fine but we do not want to log when are re-throwing an exception.